### PR TITLE
Fix IOP bltzal and bgezal opcodes

### PR DESCRIPTION
**Pull request by Kingcom**
Turns out branches that link do always link, even if the branch is not taken.

### CODE REVIEW

File: pcsx2/R3000AInterpreter.cpp [hunk 1]
```diff
@@ -39,9 +39,9 @@ void psxBGEZ()         // Branch if Rs >= 0
 
 void psxBGEZAL()   // Branch if Rs >= 0 and link
 {
+	_SetLink(31);
 	if (_i32(_rRs_) >= 0)
 	{
-		_SetLink(31);
 		doBranch(_BranchTarget_);
```
Comment (CalmCodeC) [inline pcsx2/R3000AInterpreter.cpp]: why not use ``_SetLink`` ``after doBranch()`` to have it follow the order as mentioned on comment ?

Comment (Kingcom) [inline pcsx2/R3000AInterpreter.cpp]: The branch changes the PC, _SetLink needs to work with the old PC. The comment is mostly just the opcode name written out.

Comment (CalmCodeC) [inline pcsx2/R3000AInterpreter.cpp]: store the old PC before branch and do a ``_SetLink()`` afterwards ? I think it's worth it to prevent confusion.

Comment (Kingcom) [inline pcsx2/R3000AInterpreter.cpp]: The branch executes the delay slot, which may change RA. Setting RA after the branch would incorrectly overwrite that change.

Comment (sudonim1) [inline pcsx2/R3000AInterpreter.cpp]: It's not confusing at all.

**Pull request merged**